opencti-platform/opencti-front/src/private/components/common/lists/FilterAutocomplete.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { Dispatch, FunctionComponent, SyntheticEvent, useState } from 'react';
+import TextField from '@mui/material/TextField';
+import * as R from 'ramda';
+import Autocomplete from '@mui/material/Autocomplete';
+import makeStyles from '@mui/styles/makeStyles';
+import { AutocompleteValue } from '@mui/material';
+import ItemIcon from '../../../../components/ItemIcon';
+import { useFormatter } from '../../../../components/i18n';
+import useSearchEntities from '../../../../utils/filters/useSearchEntities';
+import { Theme } from '../../../../components/Theme';
+import SearchScopeElement from './SearchScopeElement';
+import { onlyGroupOrganization } from '../../../../utils/filters/filtersUtils';
+import { HandleAddFilter } from '../../../../utils/hooks/useLocalStorage';
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+    color: theme.palette.primary.main,
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+}));
+
+interface OptionValue {
+  value: string
+  label: string
+  type: string
+  group?: string
+  color?: string
+}
+
+interface FilterAutocompleteProps {
+  filterKey: string
+  defaultHandleAddFilter: HandleAddFilter
+  inputValues: Record<string, string | Date>
+  setInputValues: Dispatch<Record<string, string | Date>>
+  availableEntityTypes: Array<string>
+  availableRelationshipTypes: Array<string>
+  allEntityTypes: boolean
+  openOnFocus: boolean
+}
+
+const FilterAutocomplete: FunctionComponent<FilterAutocompleteProps> = ({
+  filterKey,
+  defaultHandleAddFilter,
+  inputValues,
+  setInputValues,
+  availableEntityTypes,
+  availableRelationshipTypes,
+  allEntityTypes,
+  openOnFocus,
+}) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+
+  const [searchScope, setSearchScope] = useState<Record<string, Array<string>>>({});
+
+  const [entities, searchEntities]: [Record<string, Array<OptionValue>>, (filterKey: string, event: SyntheticEvent) => Record<string, Array<OptionValue>>] = useSearchEntities({
```

Comment:
Either this declaration or the as is useless

Reply:
I think you could have a useSearchEntities.d.ts declaring the type of the file in js

Reply:
Fix but I don't add a declaring type for now